src/engraving/libmscore/noteentry.cpp

```diff
@@ -162,14 +162,22 @@ Note* Score::addPitch(NoteVal& nval, bool addFlag, InputState* externalInputStat
         bool isTied = (c
                        && c->isChord()
                        && !toChord(c)->notes().empty()
-                       && (toChord(c)->notes()[0]->tieFor() || toChord(c)->notes()[0]->tieBack()));
+                       && (toChord(c)->notes().at(0)->tieFor() || toChord(c)->notes().at(0)->tieBack()));
```

Comment:
This code should be rewritten. We call toChord(c)->notes()/toChord(c)->notes().at(0) many times. We should only call each operation once